### PR TITLE
Create proc_creation_macos_enable_root_account.yml

### PR DESCRIPTION
**Pull request by D4rkCiph3r**
## Summary of the Pull Request
The pull request adds a new rule for macOS (T1078, T1078.001)

## Detailed Description of the Pull Request / Additional comments

The rule helps detect attempts to enable/add an account to the admin group, thus granting the root privilege using various utilities such as dsenableroot

## Example Log Event (In Case of FP Fixes)
NA

## Relevant Issues (In Case of Issue Fixes)
NA

### CODE REVIEW

File: rules/macos/process_creation/proc_creation_macos_enable_root_account.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+title: Enable the Root Account
+id: 821bcf4d-46c7-4b87-bc57-9509d3ba7c11
+status: experimental
+description: Detects attempts to enable/add an account to the admin group, thus granting the root privilege
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1078.001/T1078.001.md
```
Comment (frack113) [inline rules/macos/process_creation/proc_creation_macos_enable_root_account.yml:6]:
```suggestion
```
Get only windows tests.

File: rules/macos/process_creation/proc_creation_macos_enable_root_account.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    category: process_creation
+    product: macos
+detection:
+    selection1:
+        Image|endswith:
+            - '/dseditgroup'
+            - '/dscl'
+    selection2:
+        CommandLine|contains:
+            - '/Groups/admin'
```
Comment (frack113) [inline rules/macos/process_creation/proc_creation_macos_enable_root_account.yml:28]: '/Groups/admin' will be cover by 'admin'

**Comment by frack113** [comment]
It more easy to split '/dseditgroup' and '/dscl'
In the rule, the options are mix and you need to read the doc to find the correct one.

File: rules/macos/process_creation/proc_creation_macos_enable_root_account.yml (hunk 2 above)
Comment (D4rkCiph3r) [reply rules/macos/process_creation/proc_creation_macos_enable_root_account.yml:27]: Updated the rule!

File: rules/macos/process_creation/proc_creation_macos_enable_root_account.yml (hunk 1 above)
Comment (D4rkCiph3r) [reply rules/macos/process_creation/proc_creation_macos_enable_root_account.yml]: Removed and added a correct reference! (Btw, the PR for T1078.003 ART is yet to be approved for other tests)

File: rules/macos/process_creation/proc_creation_macos_enable_root_account.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        Image|endswith: '/dseditgroup'
+        CommandLine|contains|all:
+            - '-a' #name of the record(username)
+            - '-t' #type of the record(usergroup)
+            - 'admin'
+    selection_dsenableroot:
+        Image|endswith: '/dsenableroot'
+    filter:
+        CommandLine|contains: ' -d '
+    condition: (selection_dsenableroot and not filter) or 1 of selection*
```
Comment (frack113) [inline rules/macos/process_creation/proc_creation_macos_enable_root_account.yml:38]: 'selection*' will cover selection_dsenableroot too.
```suggestion
    condition: (selection_root and not filter_root) or 1 of selection_ds_*
```
selection_dscl => selection_ds_cl
selection_dseditgroup => selection_ds_editgroup
selection_dsenableroot => selection_root
filter => filter_root

**Review by frack113 — APPROVED** [review]

**Review by nasbench — APPROVED** [review]

**Review by phantinuss — APPROVED** [review]

**Pull request merged**